locust/test/test_dispatch.py

```diff
@@ -3291,6 +3291,241 @@ class User3(User):
+            [(1, 1, 1, 1, 1), (200, 0, 0, 0, 0)],
+            [(1, 1, 1, 1, 1), (0, 0, 0, 0, 0)],
+        ]:
+
+            u1_weight, u2_weight, u3_weight, u4_weight, u5_weight = weights
+            u1_fixed_count, u2_fixed_count, u3_fixed_count, u4_fixed_count, u5_fixed_count = fixed_counts
+
+            class User1(User):
+                weight = u1_weight
+                fixed_count = u1_fixed_count
+
+            class User2(User):
+                weight = u2_weight
+                fixed_count = u2_fixed_count
+
+            class User3(User):
+                weight = u3_weight
+                fixed_count = u3_fixed_count
+
+            class User4(User):
+                weight = u4_weight
+                fixed_count = u4_fixed_count
+
+            class User5(User):
+                weight = u5_weight
+                fixed_count = u5_fixed_count
+
+            target_user_counts = [sum(fixed_counts), sum(fixed_counts) + 100]
+            down_counts = [0, max(min(fixed_counts) - 1, 0)]
+            user_classes = [User1, User2, User3, User4, User5]
+
+            for worker_count in [3, 5, 9]:
+                workers = [WorkerNode(str(i + 1)) for i in range(worker_count)]
+                users_dispatcher = UsersDispatcher(worker_nodes=workers, user_classes=user_classes)
+
+                for down_to_count in down_counts:
+                    for target_user_count in target_user_counts:
+
+                        # Ramp-up to go to `target_user_count` #########
+
+                        users_dispatcher.new_dispatch(target_user_count=target_user_count, spawn_rate=1)
+                        users_dispatcher._wait_between_dispatch = 0
+
+                        list(users_dispatcher)
+
+                        for user_class in user_classes:
+                            if user_class.fixed_count:
+                                self.assertEqual(
+                                    users_dispatcher._get_user_current_count(user_class.__name__),
+                                    user_class.fixed_count,
+                                )
+
+                        # Ramp-down to go to `down_to_count`
+                        # and ensure what the fixed users was decreased too
+
+                        users_dispatcher.new_dispatch(target_user_count=down_to_count, spawn_rate=1)
+                        users_dispatcher._wait_between_dispatch = 0
+
+                        list(users_dispatcher)
+
+                        for user_class in user_classes:
+                            if user_class.fixed_count:
+                                self.assertNotEqual(
+                                    users_dispatcher._get_user_current_count(user_class.__name__),
+                                    user_class.fixed_count,
+                                )
+
+                        # Ramp-up go back to `target_user_count` and ensure
+                        # what the fixed users return to their counts
```

Comment:
Unless I misunderstood the sentence, I think the "what" does not belong in that sentence:
```suggestion
                        # the fixed users return to their counts
```

Reply:
I guess this subject jumped from my native language and not needed in eng ))) Removed